libs/api/ebpf_api.cpp

```diff
@@ -1753,11 +1753,17 @@ _initialize_ebpf_object_from_native_file(
     }
 
 Exit:
-    ebpf_free(program);
     if (result != EBPF_SUCCESS) {
+        if (program && object.programs.size() == 0) {
```

Comment:
question: Why do we need to check `object.programs.size()`?
suggestion: we can instead use `clean_up_ebpf_program()` to clean up program. But we will need to set `program->fd = ebpf_fd_invalid;` above when initializing.

Reply:
1. The program can be nullptr, at the below exit. Hence the program check is added, before derefencing problem at Exit. 

_initialize_ebpf_object_from_native_file
{
    ebpf_program_t* program = nullptr;

    EBPF_LOG_ENTRY();
    ebpf_assert(file_name);
    ebpf_assert(error_message);

    ebpf_section_info_t* infos = nullptr;
    ebpf_result_t result = _ebpf_enumerate_native_sections(file_name, &object, pin_root_path, &infos, error_message);
    **if (result != EBPF_SUCCESS) {
        goto Exit;     <<<<< 1. 
    }**

2. program is still not added to object map. 
        program->section_name = ebpf_duplicate_string(info->section_name);
        if (program->section_name == nullptr) {
            result = EBPF_NO_MEMORY;  
            goto Exit;
        }

        program->program_name = ebpf_duplicate_string(info->program_name);
        if (program->program_name == nullptr) {
            result = EBPF_NO_MEMORY;     <<<< Failure, hence free the above program->section_name.
            goto Exit;
        }

        // Update attach type for the program.
        if (get_global_program_type() != nullptr) {
            const ebpf_attach_type_t* attach_type = get_global_attach_type();
            if (attach_type != nullptr) {
                program->attach_type = *attach_type;
            }
        }

        object.programs.emplace_back(program);    <<<< the code is not reached here.

Reply:
Thanks. Fixed as suggested.